client-js/test/spec/tsTestSpec.ts

```diff
@@ -0,0 +1,42 @@
+import {Agent} from '../test/helloWorld';
+import {Observable} from '../app/scripts/observable';
+
+
+describe("Hello world", function() {
+  var agent;
+
+  beforeEach(function() {
+    agent = new Agent();
+  });
+
+  it("says hello", function() {
+    expect(agent.helloWorld()).toEqual("Hello world!");
+  });
+});
+
+/*
+* Test to create new observable and add callback
+* This is mostly to see if the plumbing for the
+* tests interfaces with the scripts in /app
+*/
+
+describe("observable", function() {
+  var observable;
+
+  beforeEach(function() {
+    observable = new Observable<any>();
+
+  });
+
+  function testFunc(message: string) {
+    console.log("testFunc called");
+    return message;
+  }
+
+
+  it("Add Callback", function() {
+    console.log("In callback test");
+    expect(observable.add(testFunc("something"))).toEqual("something");
```

Comment:
I think I see the problem. You need to actually declare the type of the `observable` variable. Right now it is implicitly typed as `any`. Instead, you should do:
```javascript
var observable: Observable<any>;

beforeEach(() => {
  observable = new Observable<any>();
});

...
```
I don't know what type the compiler was inferring for this, but that should help.

Of course, then you'll have to the test itself. Calling `.add` itself does not return anything. You'll want to actually call the observable and check that the callback was invoked.